servers/rendering/rasterizer_rd/shaders/particles.glsl

```diff
@@ -217,6 +246,206 @@ void main() {
+						//point outside box
+
+						vec3 closest = min(abs_pos, FRAME.colliders[i].extents);
+						vec3 rel = abs_pos - closest;
+						depth = length(rel) - particle_size;
+						if (depth < 0.0) {
+							col = true;
+							normal = mat3(FRAME.colliders[i].transform) * (normalize(rel) * sgn_pos);
+							depth = -depth;
+						}
+					} else {
+						//point inside box
+						vec3 axis_len = FRAME.colliders[i].extents - abs_pos;
+						// there has to be a faster way to do this?
+						if (all(lessThan(axis_len.xx, axis_len.yz))) {
+							normal = vec3(1, 0, 0);
+						} else if (all(lessThan(axis_len.yy, axis_len.xz))) {
+							normal = vec3(0, 1, 0);
+						} else {
+							normal = vec3(0, 0, 1);
+						}
+
+						col = true;
+						depth = dot(normal * axis_len, vec3(1)) + particle_size;
+						normal = mat3(FRAME.colliders[i].transform) * (normal * sgn_pos);
+					}
+
+				} break;
+				case COLLIDER_TYPE_SDF: {
+					vec3 apos = abs(local_pos);
+					float extra_dist = 0.0;
+					if (any(greaterThan(apos, FRAME.colliders[i].extents))) { //outside
+						vec3 mpos = min(apos, FRAME.colliders[i].extents);
+						extra_dist = distance(mpos, apos);
+					}
+
+					if (extra_dist > particle_size) {
+						continue;
+					}
+
+					vec3 uvw_pos = (local_pos / FRAME.colliders[i].extents) * 0.5 + 0.5;
+					float s = texture(sampler3D(sdf_vec_textures[FRAME.colliders[i].texture_index], material_samplers[SAMPLER_LINEAR_CLAMP]), uvw_pos).r;
+					s *= FRAME.colliders[i].scale;
+					s += extra_dist;
+					if (s < particle_size) {
+						col = true;
+						depth = particle_size - s;
+						const float EPSILON = 0.001;
+						normal = mat3(FRAME.colliders[i].transform) * normalize(vec3(
+																			  texture(sampler3D(sdf_vec_textures[FRAME.colliders[i].texture_index], material_samplers[SAMPLER_LINEAR_CLAMP]), uvw_pos + vec3(EPSILON, 0.0, 0.0)).r - texture(sampler3D(sdf_vec_textures[FRAME.colliders[i].texture_index], material_samplers[SAMPLER_LINEAR_CLAMP]), uvw_pos - vec3(EPSILON, 0.0, 0.0)).r,
+																			  texture(sampler3D(sdf_vec_textures[FRAME.colliders[i].texture_index], material_samplers[SAMPLER_LINEAR_CLAMP]), uvw_pos + vec3(0.0, EPSILON, 0.0)).r - texture(sampler3D(sdf_vec_textures[FRAME.colliders[i].texture_index], material_samplers[SAMPLER_LINEAR_CLAMP]), uvw_pos - vec3(0.0, EPSILON, 0.0)).r,
+																			  texture(sampler3D(sdf_vec_textures[FRAME.colliders[i].texture_index], material_samplers[SAMPLER_LINEAR_CLAMP]), uvw_pos + vec3(0.0, 0.0, EPSILON)).r - texture(sampler3D(sdf_vec_textures[FRAME.colliders[i].texture_index], material_samplers[SAMPLER_LINEAR_CLAMP]), uvw_pos - vec3(0.0, 0.0, EPSILON)).r));
+					}
+
+				} break;
+				case COLLIDER_TYPE_HEIGHT_FIELD: {
+					vec3 local_pos_bottom = local_pos;
+					local_pos_bottom.y -= particle_size;
+
+					if (any(greaterThan(abs(local_pos_bottom), FRAME.colliders[i].extents))) {
+						continue;
+					}
+
+					const float DELTA = 1.0 / 8192.0;
```

Comment:
Shouldnt this be 1.0/ size_of_height_field_texture?

Reply:
problem is that the texture is not really po2 or square, maybe should just pass this a a vec2